### PR TITLE
modulesync 5.3 & update EoL URI syntax + a lot of rubocop rework

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
this is mostly rubocop -a (safe autofix) and a lot of rework. Since this module contains a lot of types and providers, please review this carefully.

File: spec/acceptance/authentication_spec.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
+# frozen_string_literal: true
+
 require 'spec_helper_acceptance'
 require 'uri'
 
 context 'authenticated download' do
   let(:source) do
-    URI.escape("http://httpbin.org/basic-auth/user/#{password}")
+    CGI.escape("http://httpbin.org/basic-auth/user/#{password}")
```
Comment (root-expert) [inline spec/acceptance/authentication_spec.rb:8]: `CGI.escape` doesn't play well with `URI.regexp` (https://github.com/voxpupuli/puppet-archive/blob/master/lib/puppet/type/archive.rb#L127-L129)

I'm wondering if switching to normal regexp is an option?

Comment (bastelfreak) [reply spec/acceptance/authentication_spec.rb:8]: I am fine with whatever works :D

**Comment by prolixalias** [comment]
Hey there Tim... I resurrected this PR moments ago, rebased from a recent PR I'd submitted. Once it's through, is it of any use to you for me to try and wrap up #463 ?

**Comment by bastelfreak** [comment]
@prolixalias thanks for the work, can you take a look at the failing tests?

**Comment by bastelfreak** [comment]
It looks like all acceptance tests are failling, not only arch linux. because of:
```
Failure/Error: URI.escape("http://httpbin.org/basic-auth/user/#{password}") # rubocop:disable Lint/UriEscapeUnescape
NoMethodError:
  undefined method `escape' for URI:Module
```

I'm not sure whats the proper fix here. @alexjfisher I think you solved that in another module recently?

**Comment by prolixalias** [comment]
> It looks like all acceptance tests are failling, not only arch linux. because of:
> 
> ```
> Failure/Error: URI.escape("http://httpbin.org/basic-auth/user/#{password}") # rubocop:disable Lint/UriEscapeUnescape
> NoMethodError:
>   undefined method `escape' for URI:Module
> ```
> 
> I'm not sure whats the proper fix here. @alexjfisher I think you solved that in another module recently?

It's fixed in my [PR475](https://github.com/voxpupuli/puppet-archive/pull/475/checks) Figured I'd rebase your branch from master when everything made it through.

**Pull request merged**